flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/sink/KafkaSerializerWrapper.java

```diff
@@ -32,26 +32,46 @@
 import static org.apache.flink.util.Preconditions.checkNotNull;
 import static org.apache.flink.util.Preconditions.checkState;
 
+/** Wrapper for Kafka {@link Serializer}. */
 class KafkaSerializerWrapper<IN> implements SerializationSchema<IN> {
+
     private final Class<? extends Serializer<? super IN>> serializerClass;
+    // whether the serializer is for key or value
+    private final boolean isKey;
     private final Map<String, String> config;
     private final Function<? super IN, String> topicSelector;
 
     private transient Serializer<? super IN> serializer;
 
     KafkaSerializerWrapper(
             Class<? extends Serializer<? super IN>> serializerClass,
+            boolean isKey,
             Map<String, String> config,
             Function<? super IN, String> topicSelector) {
         this.serializerClass = checkNotNull(serializerClass);
+        this.isKey = isKey;
         this.config = checkNotNull(config);
         this.topicSelector = checkNotNull(topicSelector);
     }
 
     KafkaSerializerWrapper(
             Class<? extends Serializer<? super IN>> serializerClass,
+            Map<String, String> config,
+            Function<? super IN, String> topicSelector) {
+        this(serializerClass, false, config, topicSelector);
+    }
+
+    KafkaSerializerWrapper(
```

Comment:
nits: Previously we have two constructors here. Now we have 4 constructors due to the addition of `isKey` parameter. 

Would it be simpler to keep only two constructors and let caller always explicitly provides the `isKey` value? This probably also makes the `KafkaRecordSerializationSchemaBuilder` code more readable.